gobblin-core/src/main/java/gobblin/source/extractor/partition/Partitioner.java

```diff
@@ -237,6 +233,7 @@ public Partitioner(SourceState state) {
    * @return the adjusted watermark value
    */
   synchronized private static long adjustWatermark(String baseWatermark, WatermarkType watermarkType, int diff) {
+    SimpleDateFormat watermarkTimeParser = new SimpleDateFormat(WATERMARKTIMEFORMAT);
```

Comment:
Removed the synchronized, and yes, I am using the gobblin code style here https://gobblin.readthedocs.io/en/latest/developer-guide/files/codestyle-intellij-gobblin.xml